cfgrammar/src/lib/yacc/parser.rs

```diff
@@ -1794,4 +1819,30 @@ x"
             Some(("a::b".to_owned(), "(u64, u64)".to_owned()))
         );
     }
+
+    #[test]
+    fn test_error_kinds() {
+        macro_rules! expect_yacc_error_kind {
+            ($yacc_kind:path, $src:expr, $error_kind:path) => {{
+                match parse($yacc_kind, $src) {
+                    Err(YaccParserError {
+                        kind: $error_kind, ..
+                    }) => (),
+                    Err(e) => panic!("incorrect error {:?} expected error {:?}", e, $error_kind),
+                    Ok(_) => panic!("parsed successfully when expecting error {:?}", $error_kind),
+                }
+            }};
+        }
+
+        expect_yacc_error_kind!(
+            YaccKind::Grmtools,
+            "%start A
+             %%
+             A -> (): 'a1';
+             A -> (): 'a2';
+             A -> (): 'a3';
+            ",
+            YaccParserErrorKind::DuplicateRule
```

Comment:
Yeah, it can't yet because we only ever return the one of these errors from `parse_rules`.
I've a local `assert_eq!(errors.len(), 2);` check, but that check really doesn't belong there.

We figure out the error type to return multiple things, before we can do that.